### PR TITLE
Add an optional password strength meter.

### PR DESCRIPTION
**Pull request by realityking**

### CODE REVIEW

**Comment by eddieajau** [comment]
Just want to let you know this looks really good, but I just haven't had time to do a live test of it yet.

**Comment by ianmacl** [comment]
# Test Results

Total Tests: 1280
Assertions: 2217
Failures: 0
Errors: 0
Test Time: 56.872088

**Comment by ianmacl** [comment]
# Test Results

Total Tests: 
Assertions: 
Failures: 
Errors: 
Test Time:

**Comment by ianmacl** [comment]
# Test Results

Total Tests: 1420
Assertions: 9898
Failures: 0
Errors: 0
Test Time: 167.986629

**Comment by ianmacl** [comment]
# Test Results

Total Tests: 1465
Assertions: 10002
Failures: 0
Errors: 0
Test Time: 196.620230

File: libraries/joomla/form/fields/password.php [hunk 1]
```diff
@@ -41,17 +41,36 @@ class JFormFieldPassword extends JFormField
...
-		$readonly = ((string) $this->element['readonly'] == 'true') ? ' readonly="readonly"' : '';
-		$disabled = ((string) $this->element['disabled'] == 'true') ? ' disabled="disabled"' : '';
+		$size		= $this->element['size'] ? ' size="'.(int) $this->element['size'].'"' : '';
+		$maxLength	= $this->element['maxlength'] ? ' maxlength="'.(int) $this->element['maxlength'].'"' : '';
+		$class		= $this->element['class'] ? ' class="'.(string) $this->element['class'].'"' : '';
+		$auto		= ((string) $this->element['autocomplete'] == 'off') ? ' autocomplete="off"' : '';
+		$readonly	= ((string) $this->element['readonly'] == 'true') ? ' readonly="readonly"' : '';
+		$disabled	= ((string) $this->element['disabled'] == 'true') ? ' disabled="disabled"' : '';
+		$meter		= ((string) $this->element['strengthmeter'] == 'true');
+		$treshold	= $this->element['treshold'] ? (int) $this->element['treshold'] : 66;
```
Comment (chdemko) [inline libraries/joomla/form/fields/password.php]: Could you respect correct spelling for threshold instead of 'treshold'

Comment (realityking) [inline libraries/joomla/form/fields/password.php]: Done. You actually found a bug in the process ;)

**Pull request merged**